### ISSUE TITLE
Could not run parameterized with DBUnitRule with database rider > 1.2.9

### ISSUE THREAD

**Issue by JoergSiebahn** [database-rider/database-rider#104]
Lookup for method annotations fails with parametrized tests in DBUnitRule.apply:

```
java.lang.RuntimeException: Could not find method shouldJustRunMoreThanOneTime[0] on test class ParameterizedTest 

	at com.github.database.rider.core.util.AnnotationUtils.findAnnotation(AnnotationUtils.java:179)
	at com.github.database.rider.core.DBUnitRule.apply(DBUnitRule.java:57)
        ...
```

Running parameterized tests is fine with version 1.2.9 but fails with all later versions including 1.3.0 and 1.4.0

You may refer to this test case for verification:

```
@RunWith(Parameterized.class)
@DBUnit(url = "jdbc:h2:mem:test",  driver = "org.h2.Driver", user = "u", password = "p")
public class ParameterizedTest {

   private int given;
   private int expected;

   @Rule
   public final DBUnitRule dbUnitRule = DBUnitRule.instance();

   @Parameters
   public static Collection<Object[]> data() {
      return Arrays.asList(
            new Object[] {0,0},
            new Object[] {1,1},
            new Object[] {2,2}
      );
   }

   public ParameterizedTest(int given, int expected) {
      this.given = given;
      this.expected = expected;
   }

   @Test
   public void shouldJustRunMoreThanOneTime() {
      Assertions.assertThat(given).isEqualTo(expected);
   }
}
```

**Comment by rmpestano** [comment]
Hi @JoergSiebahn, 

After 2 years I'm not sure you're still using dbrider but if so, can you try using v`1.18.1-SNAPSHOT`? 

You'll need to enable snapshots repo on your build system:

```
<repositories>
    <repository>
        <snapshots/>
        <id>snapshots</id>
        <name>libs-snapshot</name>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```

**Comment by JoergSiebahn** [comment]
Thanks for coming back to this issue.

We still suggest to use dbrider for SQL DBs in [our library](https://github.com/SDA-SE/sda-dropwizard-commons) that is now open source. But we were stuck on the old version. Honestly it is not in very active use because most of our services use MongoDB. So I can't test the version in detail.

I tried 1.18.1-SNAPSHOT together with DBUnit 2.5.3 and the latest 2.7.0 as well.
The [parameterized test](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-hibernate-testing/src/test/java/org/sdase/commons/server/hibernate/testing/ParameterizedTest.java) passes in both cases.

But there is [another test](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-hibernate/src/test/java/org/sdase/commons/server/hibernate/HibernateIT.java) (without parameterized) [failing at class level](https://github.com/SDA-SE/sda-dropwizard-commons/pull/699/checks?check_run_id=1537996410#step:4:812):

```
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.lang.Class.getMethod(Class.java:2102)
	at com.github.database.rider.core.util.AnnotationUtils.findAnnotation(AnnotationUtils.java:169)
	at com.github.database.rider.core.DBUnitRule.apply(DBUnitRule.java:56)
	at org.junit.rules.RunRules.applyAll(RunRules.java:26)
	at org.junit.rules.RunRules.<init>(RunRules.java:15)
	at org.junit.runners.ParentRunner.withClassRules(ParentRunner.java:269)
	at org.junit.runners.ParentRunner.classBlock(ParentRunner.java:217)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:412)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
```

I prepared a [PR](https://github.com/SDA-SE/sda-dropwizard-commons/pull/699) for the upgrade.

**Comment by rmpestano** [comment]
Hi @JoergSiebahn, thank you very much for the feedback, the NPE should be fixed now. Let me know and then we can release a new version.

Thank you again!

**Comment by rmpestano** [comment]
Hi again, the fix is available in just released `1.19.0`. If the issue persists just reopen this issue.

Thanks for the feedback.

**Comment by JoergSiebahn** [comment]
Thank you very much. We integrated the latest version successfully.

**Comment by victor-costa-avancee** [comment]
Is it just me or is something similar happening in current version (1.24.0)?
```
    @ParameterizedTest
    @ValueSource(longs = {
            1L, //OwnerGroup Admin
            2L  //OwnerGroup User
    })
    @DataSet(value = "datasets/import.yml")
    void gateNumericAction(long userId) {
       ....
```
I am forced to add the method annotation @DataSet although I have the exact same annotation on the test class

**Comment by rmpestano** [comment]
Hi again @victor-costa-avancee,  we have this parameterized test which has `@DataSet` only at class level:
https://github.com/database-rider/database-rider/blob/d94ac85368d24737f280df88edea97e047d86dfd/rider-junit5/src/test/java/com/github/database/rider/junit5/ParameterizedClassLevelIt.java#L15-L27

Is this the same you're trying or there's something else?

**Comment by victor-costa-avancee** [comment]
Im using rider-cdi. Here is a reproducer https://github.com/victor-costa-avancee/DBRider-param-reproducer

**Comment by rmpestano** [comment]
Hi again @victor-costa-avancee, I couldn't run your sample because of database setup so I tried to reproduce on quarkus dbunit sample which uses CDI but It's working there, can you have a look? https://github.com/database-rider/database-rider/blob/3e0aed1a149cf9d152e7229cb08c4aba373acd7d/rider-examples/quarkus-dbunit-sample/src/test/java/com/github/quarkus/sample/QuarkusDBUnitParameterizedTest.java#L32-L55

I hope It helps

**Comment by victor-costa-avancee** [comment]
I didn't test it on the H2 DB but if you remove the following lines from the application.properties file:
```#%test.quarkus.datasource.jdbc.url = jdbc:h2:mem:cdbookstoreDB
#%test.quarkus.datasource.jdbc.driver = org.h2.Driver```
So that tests will run using Postgres, both `QuarkusDBUnitParameterizedTest.shouldFindBookById` and `QuarkusDBUnitParameterizedTest.shouldFindBookByTitle`fail.

**Comment by victor-costa-avancee** [comment]
I also had to fix the username and password configuration, the properties should be (without the JDBC part)
```
quarkus.datasource.username
quarkus.datasource.password
```

**Comment by victor-costa-avancee** [comment]
I managed to spend some more time on this and tested with the H2 DB. If you comment out the `shouldFindBookByTitleUsingMethodLevelDataSet` test, the other two will throw an AssertionError.

**Comment by rmpestano** [comment]
Hey, good catch, now I'm able to reproduce and fix. Thank you!

**Comment by rmpestano** [comment]
Can you try `1.25.1-SNAPSHOT` @victor-costa-avancee ? `You'll have to enable snapshots repo on your build, [see here](https://github.com/database-rider/database-rider#snapshots).

**Comment by victor-costa-avancee** [comment]
It seems to be working perfectly @rmpestano, great job, thank you!

**Comment by rmpestano** [comment]
Thank you! This fix is available in the just released v`1.26.0`.
